packages/react-scripts/package.json

```diff
@@ -62,6 +63,7 @@
     "sass-loader": "7.1.0",
     "style-loader": "0.23.0",
     "terser-webpack-plugin": "1.1.0",
+    "typescript": "3.1.1",
```

Comment:
We still need to figure out a way to remove this before this can be merged.

Reply:
I recommend we remove this as a dependency, and allow the consumer to control the version of typescript, which looks to be in line with [flow support](https://github.com/facebook/create-react-app/blob/9fd98447d8afeef90efc669040b52fc65bc06484/packages/react-scripts/template/README.md#adding-flow). It was a constant source of churn in https://github.com/wmonk/create-react-app-typescript (every time a new version of typescript came out, there were people posting to update the dependency). I changed to use a peerDependency in that case and it has worked out quite well!

Reply:
I need help with that. I've tried simply removing it and installing on the user app side, and also tried adding to peerDependencies, but `fork-ts-checker-webpack-plugin` keeps failing to find it:

```
Error: Cannot find module 'typescript'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/brunolemos/Projects/packages/create-react-app/node_modules/fork-ts-checker-webpack-plugin/lib/CancellationToken.js:6:10)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
```

I opened this issue: https://github.com/Realytics/fork-ts-checker-webpack-plugin/issues/165

Reply:
Done https://github.com/facebook/create-react-app/pull/4837#issuecomment-428506877